pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1764,12 +1800,36 @@ BEGIN
                     ORDER BY d.id ASC
                     LIMIT 1
                 ) as chunk_interval,
-                ARRAY(
-                    SELECT key
-                    FROM SCHEMA_CATALOG.label_key_position lkp
-                    WHERE lkp.metric_name = m.metric_name
-                    ORDER BY key) label_keys,
-                hi.total_size as size,
+                (
+                    SELECT COALESCE(
+                         SUM(
+                              UPPER(rs.ranges[1]::TSTZRANGE) - LOWER(rs.ranges[1]::TSTZRANGE)
+                         ) FILTER (WHERE cs.compression_status = 'Compressed'),
+                         INTERVAL '0'
+                         ) compressed_interval
+                    FROM
+                        public.chunk_relation_size_pretty(FORMAT('SCHEMA_DATA.%I', m.table_name)) rs
+                    LEFT JOIN
+                        timescaledb_information.compressed_chunk_stats cs
+                    ON (cs.chunk_name::text = rs.chunk_table::text)
+                ) as compressed_interval,
+                (
+                    SELECT COALESCE(
+                         SUM(
+                            UPPER(rs.ranges[1]::TSTZRANGE) - LOWER(rs.ranges[1]::TSTZRANGE)
+                        ),
+                        INTERVAL '0'
+                        ) total_interval
+                    FROM
+                        public.chunk_relation_size_pretty(FORMAT('SCHEMA_DATA.%I', m.table_name)) rs
+                    LEFT JOIN
+                        timescaledb_information.compressed_chunk_stats cs
+                    ON (cs.chunk_name::text = rs.chunk_table::text)
+                ) as total_interval,
```

Comment:
This should be one subquery instead of two for efficiency
```select                                                                                                                                                                                 coalesce(sum(upper(rs.ranges[1]::tstzrange) - lower(rs.ranges[1]::tstzrange)), interval '0') total_interval,                                                                                                    coalesce(sum(upper(rs.ranges[1]::tstzrange) - lower(rs.ranges[1]::tstzrange)) filter (where cs.compression_status = 'Compressed'), interval '0'                                                          ) compressed_interval                                                                                                                                                                                          FROM public.chunk_relation_size_pretty('prom_data.go_goroutines') rs                                                                                                                                            LEFT JOIN timescaledb_information.compressed_chunk_stats cs on (cs.chunk_name::text = rs.chunk_table::text)```

Reply:
The reason I did is, subquery must return only one value. I even tried the left join thing, and it was even slower. So went with this.

Reply:
The current one takes 4 secs, and the one I did with single subquery as left join (below) took 14 secs

```
SELECT
                m.id,
                m.metric_name,
                m.table_name,
                ARRAY(
                    SELECT key
                    FROM _prom_catalog.label_key_position lkp
                    WHERE lkp.metric_name = m.metric_name
                    ORDER BY key) label_keys,
                _prom_catalog.get_metric_retention_period(m.metric_name) as retention_period,
                (
                    SELECT _timescaledb_internal.to_interval(interval_length)
                    FROM _timescaledb_catalog.dimension d
                    WHERE d.hypertable_id = h.id
                    ORDER BY d.id ASC
                    LIMIT 1
                ) as chunk_interval,
                ci.compressed_interval,
                ci.total_interval,
                pg_size_bytes(chs.uncompressed_total_bytes) as before_compression_bytes,
                pg_size_bytes(chs.compressed_total_bytes) as after_compression_bytes,
                pg_size_bytes(hi.total_size) as total_bytes,
                hi.total_size as total_size,
                (1.0 - (pg_size_bytes(chs.compressed_total_bytes)::numeric / pg_size_bytes(chs.uncompressed_total_bytes)::numeric)) * 100 as compression_ratio,
                chs.total_chunks,
                chs.number_compressed_chunks as compressed_chunks
            FROM _prom_catalog.metric m
            LEFT JOIN timescaledb_information.hypertable hi ON
                        (hi.table_schema = 'prom_data' AND hi.table_name = m.table_name)
            LEFT JOIN timescaledb_information.compressed_hypertable_stats chs ON
                        (chs.hypertable_name = format('%I.%I', 'prom_data', m.table_name)::regclass)
            LEFT JOIN _timescaledb_catalog.hypertable h ON
                        (h.schema_name = 'prom_data' AND h.table_name = m.table_name)
			LEFT JOIN (
                    SELECT COALESCE(
                         SUM(
                            UPPER(rs.ranges[1]::TSTZRANGE) - LOWER(rs.ranges[1]::TSTZRANGE)
                        ),
                        INTERVAL '0'
                        ) total_interval,
					COALESCE(
						SUM(
                              UPPER(rs.ranges[1]::TSTZRANGE) - LOWER(rs.ranges[1]::TSTZRANGE)
                         ) FILTER (WHERE cs.compression_status = 'Compressed'),
                         INTERVAL '0'
					) compressed_interval
                    FROM
                    timescaledb_information.compressed_chunk_stats cs
                left join
                    _prom_catalog.metric cm on ('prom_data.' || cm.table_name = cs.hypertable_name::text)
                left join
                    public.chunk_relation_size_pretty(FORMAT('prom_data.%I', cm.table_name)) rs
                    ON (cs.chunk_name::text = rs.chunk_table::text)
                ) as ci ON TRUE
						;
```